pkg/pillar/types/zedmanagertypes.go

```diff
@@ -276,28 +279,169 @@ type StorageStatus struct {
 }
 
 // UpdateFromStorageConfig sets up StorageStatus based on StorageConfig struct
-func (ss *StorageStatus) UpdateFromStorageConfig(sc StorageConfig) {
-	ss.DatastoreID = sc.DatastoreID
-	ss.ImageID = sc.ImageID
-	ss.Name = sc.Name
-	ss.NameIsURL = sc.NameIsURL
-	ss.ImageSha256 = sc.ImageSha256
-	ss.Size = sc.Size
-	ss.CertificateChain = sc.CertificateChain
-	ss.ImageSignature = sc.ImageSignature
-	ss.SignatureKey = sc.SignatureKey
-	ss.ReadOnly = sc.ReadOnly
-	ss.Preserve = sc.Preserve
-	ss.Format = sc.Format
-	ss.Maxsizebytes = sc.Maxsizebytes
-	ss.Devtype = sc.Devtype
-	ss.Target = sc.Target
-	if ss.Format == "container" {
-		ss.IsContainer = true
+func (ssPtr *StorageStatus) UpdateFromStorageConfig(sc StorageConfig) {
+	ssPtr.DatastoreID = sc.DatastoreID
+	ssPtr.Name = sc.Name
+	ssPtr.NameIsURL = sc.NameIsURL
+	ssPtr.ImageSha256 = sc.ImageSha256
+	ssPtr.Size = sc.Size
+	ssPtr.CertificateChain = sc.CertificateChain
+	ssPtr.ImageSignature = sc.ImageSignature
+	ssPtr.SignatureKey = sc.SignatureKey
+	ssPtr.ReadOnly = sc.ReadOnly
+	ssPtr.Preserve = sc.Preserve
+	ssPtr.Format = sc.Format
+	ssPtr.Maxsizebytes = sc.Maxsizebytes
+	ssPtr.Devtype = sc.Devtype
+	ssPtr.Target = sc.Target
+	if ssPtr.Format == "container" {
+		ssPtr.IsContainer = true
 	}
 	return
 }
 
+// IsCertsAvailable checks certificate requirement/availability for a storage object
+func (ssPtr *StorageStatus) IsCertsAvailable(safename string) (bool, error) {
```

Comment:
Since this isn't supposed to modify StorageStatus, please define it as not being a pointer i.e. ss types.StorageStatus.